src/algorithm.jl

```diff
@@ -135,7 +135,7 @@ function compute_and_save_w(phd::PHData)::Float64
             p = phd.scenario_map[s].prob
             kx = branch_value(phd, vid) - xhat
 
-            phd.scenario_map[s].w_vars[vid] += phd.r * kx
+            phd.scenario_map[s].w_vars[vid] += phd.r.value * kx
```

Comment:
For PH to work we'll need to update this computation to account for the coefficient proportionality which is actually going to be a bit complicated. I think we'll need `add_ph_objective_terms` to return a map from the `VariableID` to the used parameter value `r.value * coeff` that `_augment_subproblems` then returns via a new message type to the master process.  Then the master process has to store the values (and probably do some error checking too).

I realize this is pretty complicated and is probably a bit much since you're just learning the code.  If you don't want to do it, let me know and I can finish off the necessary pieces. However, this is a good way to learn the master/worker framework as well as the initialization order in the code.  Again, let me know if this is too much.

Reply:
I think I would like to have a stab at it. It looks like it could be a good learning experience (I'm still a bit fuzzy on how Julia does parallel computation). If I'm still lost by the end of the week, I'll hand things over to you.

Reply:
For my understanding - is it not possible to write `rho = penalty_value(phd.r, vid)` and then `rho * kx` ? Or something similar?

Reply:
Note to self: 
* `compute_and_save_w` is called by `update_ph_variables` ...
* ... is called by `hedge`...
* ... is called by `solve`

Reply:
> I think I would like to have a stab at it. It looks like it could be a good learning experience (I'm still a bit fuzzy on how Julia does parallel computation). If I'm still lost by the end of the week, I'll hand things over to you.

That's fine.  Just trying to be up front about the difficulty.  If you manage to get through it, you'll probably have a pretty good understanding of my PH code.



> For my understanding - is it not possible to write `rho = penalty_value(phd.r, vid)` and then `rho * kx` ? Or something similar?

When I first implemented this, I was following the original Progressive Hedging paper by [Rockafellar and Wets](https://pubsonline.informs.org/doi/abs/10.1287/moor.16.1.119) and so used their notation.  In their paper, the penalty parameter is `r`.  That being said, renaming it `rho` would be fine with me.

Reply:
What I meant was can't we just get the variable specific penalty parameter by calling `penalty_value(phd.r, vid)` in `compute_and_save_w`? But I realise that it's not that simple now :P